codecov_auth/authentication/repo_auth.py

```diff
@@ -398,3 +404,86 @@ def authenticate(
             RepositoryAsUser(repository),
             TokenlessAuth(repository),
         )
+
+
+class TestResultsUploadTokenRequiredAuthenticationCheck(
+    UploadTokenRequiredAuthenticationCheck
+):
+    """
+    Repository and Owner are not in the path for this endpoint, have to get them another way,
+    then use the same authenticate() as parent class.
+    """
+
+    def _get_info_from_request_data(
+        self, request: HttpRequest
+    ) -> tuple[Repository | None, Owner | None]:
+        from upload.views.test_results import UploadSerializer  # circular imports
+
+        try:
+            body = json.loads(str(request.body, "utf8"))
+        except json.JSONDecodeError:
+            return None, None  # continue to next auth class
+
+        serializer = UploadSerializer(data=body)
```

Comment:
hmmm, would it be faster/cleaner if we access the parts of the request body we care about directly instead of using the Serializer here?

Reply:
I think it's best practices to not interact with incoming data that hasn't been validated, particularly since there's no token here 😅 

I'm using the serializer to type-check the incoming fields and make sure they exist. If I didn't use the serializer, I would have to do the check that they exist manually, I'd rather have the serializer do that for me

Reply:
ok 👍